src/client/app/app.react.js

```diff
@@ -38,10 +37,12 @@ class App extends Component {
   }
 
   render() {
+    const children = React.cloneElement(this.props.children, this.state);
```

Comment:
This is bad... What about creating a HoC for the components that would wrap the component and pass the required parts of app state as props?